anker/src/token.rs

```diff
@@ -12,3 +12,4 @@ use std::{
 use lido::token::Result;
 
 impl_token!(BLamports, "bSOL");
+impl_token!(UstLamports, "UST");
```

Comment:
Beware! One of the UST tokens has 6 decimals, not 9. Using the term “lamport” would be misleading then, and the string formatter would be all wrong. But the other UST token does have 9 decimals. I don’t know which is the right one, I asked for clarification from the Terra people:

* https://solscan.io/token/9vMJfxuKxXBoEa7rM12mYLMwTacLMLDJqHozw96WQL8i (6 decimals)
* https://solscan.io/token/CXLBjMMcwkc17GfJtBos6rQCo1ypeH6eDbB82Kby4MRm (9 decimals)

Reply:
Kevin clarified that the right one is 9vMJfxuKxXBoEa7rM12mYLMwTacLMLDJqHozw96WQL8i, because that’s the one listed for the Wormhole v2 docs at https://docs.wormholenetwork.com/wormhole/overview-liquid-markets.

So we have a token with 6 decimals now :confused: We could take the format string or maybe the number of decimals as an argument to the macro, it only matters for presentation, in the end we are just putting the number of the token’s smallest denomination in a `u64`.

Reply:
Made the `impl_token` receive a decimals arg. But it's still printing 9 digits in the literal formatter `"{}.{:0>9} {}"`. Is there a nice way to pass the string literal with the alignment as `write!` parameter?

Reply:
I think we will need to pass in the format string as an argument to the macro :confused: I don’t think `std::fmt` allows the width to be chosen at runtime. There is `std::fmt::Arguments`, but it can only be constructed from a macro.

Reply:
Oh actually what we might do is just string-format the number into a `String` and then left-pad the string! Or we could format it to 9 digits and then slice the resulting &str from the right, cutting off any excess zeros on the left.